### PR TITLE
[Security] [Guard] Added type-hints to AuthenticatorInterface

### PR DESCRIPTION
**Pull request by thomasbisignani**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no  
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | #32179
| License       | MIT
| Doc PR        | 

This PR adds the type-hints to the AuthenticatorInterface

### CODE REVIEW

File: src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ abstract class AbstractGuardAuthenticator implements AuthenticatorInterface
      *
      * @return PostAuthenticationGuardToken
      */
-    public function createAuthenticatedToken(UserInterface $user, $providerKey)
+    public function createAuthenticatedToken(UserInterface $user, string $providerKey)
```
Comment (derrabus) [inline src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php]: Please also remove both `@param` annotations because they‘re redundant now.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by nicolas-grekas** [comment]
These classes contain a lot of useless `@param` annotation, could you please send a PR to remove them, on 3.4 I suppose?

**Comment by thomasbisignani** [comment]
Done here https://github.com/symfony/symfony/pull/32257 for 3.4 (@nicolas-grekas @derrabus)

File: src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php (hunk 1 above)
Comment (thomasbisignani) [reply src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php]: Given I removed the `@param UserInterface $user` annotation for 3.4 here https://github.com/symfony/symfony/pull/32257

I just removed the `@param string $provider` annotations in this PR.

**Comment by thomasbisignani** [comment]
Fixed. 
Ok for you @nicolas-grekas @derrabus ?

**Review by derrabus — APPROVED** [review]
Looks good to me, but maybe we wait until the changes of #32257 arrive in master.

**Comment by chalasr** [comment]
Rebase unlocked :)

**Comment by thomasbisignani** [comment]
Rebased.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @thomasbisignani.

**Pull request merged**